### PR TITLE
Add getLoggerIO to MonadLog class

### PR DESCRIPTION
**Pull request by arybczak**
Needed for passing Logger to libraries such as `aws` or `amazonka` that operate in the IO monad.

### CODE REVIEW

File: log-base/src/Log/Class.hs [hunk 1]
```diff
@@ -39,6 +39,8 @@ class MonadTime m => MonadLog m where
   localData   :: [Pair] -> m a -> m a
   -- | Extend the current application domain locally.
   localDomain :: T.Text -> m a -> m a
+  -- | Capture current context of a logger for use in the 'IO' monad.
```
Comment (23Skidoo) [inline log-base/src/Log/Class.hs]: I'd reword this comment as "Returns an IO action that logs messages using the current `MonadLog` context. Useful for interfacing with libraries such as `aws` or `amazonka` that accept logging callbacks that operate in IO." to make the purpose of this function more clear.

File: log-base/src/Log/Class.hs [hunk 2]
```diff
@@ -39,6 +39,8 @@ class MonadTime m => MonadLog m where
...
+  getLoggerIO :: m (UTCTime -> LogLevel -> T.Text -> Value -> IO ())
```
Comment (phadej) [inline log-base/src/Log/Class.hs]: This makes `MonadLog` unimplementable for Non-IO based monads.

I'd rather prefer separate class, e.g. `MonadLogIO` (though `NoLogT m` could still have an instance for all `m`). That wouldn't require major bump, and be a little more flexible.

Comment (arybczak) [reply log-base/src/Log/Class.hs]: Hmm, I see. I'd like to avoid additional class. How about replacing `getLoggerIO` with `getLoggerEnv :: m LoggerEnv`? It's more general as you can use it in conjunction with `logMessageIO` to get IO logging, but doesn't introduce any IO into the class.

Comment (phadej) [reply log-base/src/Log/Class.hs]: @arybczak that sounds :+1: if it works for you, though it will still be breaking change :)

**Comment by arybczak** [comment]
I updated the PR, it should be fine now.
